docs/fpc-registration-new.puml

```diff
@@ -0,0 +1,87 @@
+@startuml
+
+'- force wrapping of longer text
+'skinparam wrapWidth 400
+
+!pragma teoz true
+
+
+title Fabric Private Chaincode - Enclave registry - New approach
+
+actor	Admin order 10
+participant ECC order 20
+participant ECC_Manager order 30
+participant ERCC order 60
+
+note right ERCC
+  ERCC can be **integrated** in ECC or as **seperate** chaincode
+  Integrated:
+    - ERCC and ECC share endorsement policy 
+    - attestations stored ECC namespace
+  Separate:
+    - flexible endorsement policies (ERCC and ECC can have different)
+    - attestations stored in ERCC namespace;
+end note
+
+entity IAS order 50
+collections Orderer order 70
+collections Peers order 80
+
+group instantiation
+  Admin -> ECC : install
+  Admin -> ECC : instantiate (exp.mrenclave)
+  activate ECC
+  note right
+    With new life-cycle mrenclave defined in
+    chaincode def with approve and commit steps
+  end note
+
+  ECC -> ECC : putState(ecc.mrenclave, exp.mrenclave) 
+  ECC --> Admin : success / error
+  deactivate ECC
+end
+
+Admin -> Orderer : submit instantiate transaction
+...when next block ready...
+Orderer -> Peers : broadcast block
+Peers -> Peers : Validation
+
+group enclave creation and attestation generation
+
+  Admin -> ECC_Manager : create (CC_ID, channel)
+  activate ECC_Manager
+
+  ECC_Manager -> ECC : (CC_ID/Channel).create
+
+  create participant ECC_Enclave order 30 #99FF99 
+  ECC -> ECC_Enclave : create
+   
+  ECC_Manager -> ECC_Enclave : getQuote(SPID)
+  return quote, ECC_s_pk
+  
+  ECC_Manager -> IAS : send quote
+  activate IAS
+  IAS --> ECC_Manager : attestation
+  deactivate IAS
+
+  ECC_Manager --> Admin : attestation, ECC_s_pk / error
+  deactivate ECC_Manager
+
+end
+
+group register attestation
+  Admin -> ERCC: register(attestation, ECC_s_pk)
+  activate ERCC
+  ERCC -> ERCC : check attestation signature
```

Comment:
We probably spell out explicitly the IAS root-of-trust as this is probably also something we would want to do in the channel genesis block/definition?